services/src/datasets/upload.rs

```diff
@@ -3,15 +3,37 @@ use std::path::{Path, PathBuf};
 use crate::contexts::Session;
 use crate::error::Result;
 use crate::identifier;
+use crate::util::path_with_base_path;
 use crate::{
     error,
     util::config::{self, get_config_element},
 };
 use async_trait::async_trait;
 use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
 
 identifier!(UploadId);
 identifier!(FileId);
+identifier!(VolumeId);
+
+pub trait AdjustFilePath {
+    fn adjust_file_path(&self, file_path: &Path) -> Result<PathBuf>;
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, ToSchema)]
+pub struct Volume {
+    pub id: VolumeId,
```

Comment:
Can't we give volumes a name like in Docker, etc.?

We always have these cryptic ids that no one likes to work with.

Reply:
My thinking was that admins most likely will look for the path when selecting a volume. In docker the path is the name of the volume right? We could omit the id and only have the path, but then we would have to put some limitations on the name, e.g. length if we want to use it in URLs, permissions, etc. in the long run.

Reply:
Just allow characters, digits and hyphens or so.

In Docker, they have names:
https://docs.docker.com/storage/volumes/

So its not the path, it is a simple name like "my-vol"

Reply:
f9ce82dda228b2a2ab5852b8cab78f7748c19cd9